File: app/src/main/java/com/example/steadfast/SupplementAdapter.java
package com.example.steadfast;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.content.Context;
import android.widget.CheckBox;
import android.widget.CompoundButton;


import java.util.List;

public class SupplementAdapter extends BaseAdapter {
    private List<Supplement> mSupplements;
    private Context mContext;

    public SupplementAdapter(Context context, List<Supplement> supplements) {
        mContext = context;
        mSupplements = supplements;
    }

    @Override
    public int getCount() {
        return mSupplements.size();
    }

    @Override
    public Object getItem(int position) {
        return mSupplements.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        SupplementView view;
        if (convertView == null) {
            view = new SupplementView(mContext);
        } else {
            view = (SupplementView) convertView;
        }
        view.bind(mSupplements.get(position));

        CheckBox checkBox = (CheckBox) view.findViewById(R.id.supplement_taken);
        checkBox.setChecked(mSupplements.get(position).isTaken());
        checkBox.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                mSupplements.get(position).setTaken(isChecked);
            }
        });
        return view;
    }}

File: app/src/main/java/com/example/steadfast/SupplementAdapter.kt
package com.example.steadfast

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.steadfast.Supplement



class SupplementAdapter(private val supplements: List<Supplement>) : RecyclerView.Adapter<SupplementAdapter.ViewHolder>() {

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val nameTextView: TextView = view.findViewById(R.id.nameTextView)
        val dosageTextView: TextView = view.findViewById(R.id.dosageTextView)
        val amountTextView: TextView = view.findViewById(R.id.amountTextView)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.supplementlist, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val supplement = supplements[position]

        holder.nameTextView.text = supplement.supplement
        holder.dosageTextView.text = supplement.dosage
        holder.amountTextView.text = supplement.amount
    }

    override fun getItemCount(): Int {
        return supplements.size
    }
}
